Sources/SwiftDoubleRatchet/SwiftDoubleRatchet.swift

```diff
@@ -26,7 +16,7 @@ public struct Ratchet<C: Curve> {
         case noRecieveReceiveChain
     }
     
-    public struct Header: Codable, Equatable {
```

Comment:
You aren't using `Codable` anywhere. In fact you aren't even implementing its requirements (`init(from decoder: Decoder) throws` and `func encode(to encoder: Encoder) throws`). Instead the compiler synthesises those functions in some standard way that isn't compatible to how you do it in `init(data: Data) throws` and `var encoded: Data { get }`.